desktop/logger/namespaces/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Credit: Modified from `winston-namespace` by @SetaSouto:
+ * 	https://github.com/SetaSouto/winston-namespace
+ */
+
+module.exports = {
+  /**
+   * Boolean indicating if the object is populated with the environment data.
+   */
+  populated: false,
+  /**
+   * Populates the private data 'namespaces' as an array with the different namespaces from the DEBUG
+   * environment variable. It splits the data with ',' as separator.
+   * @private
+   */
+  populate: function () {
+    let envString = process.env.DEBUG;
+    this.namespaces = envString ? envString.split(',') : [];
+    this.populated = true;
+  },
+  /**
+   * Checks if the namespace is available to debug. The namespace could be contained in wildcards.
+   * Ex: 'server:api:controller' would pass the check (return true) if the 'server:api:controller' is in the
+   * environment variable or if 'server:api:*' or 'server:*' is in the environment variable.
+   * @param namespace {String} - Namespace to check.
+   * @returns {boolean} Whether or not the namespace is available.
+   */
+  check: function (namespace) {
+    if (!this.populated) this.populate();
+    if (this.namespaces.indexOf('*') !== -1) return true;
+    if (this.namespaces.indexOf(namespace) !== -1) return true;
```

Comment:
can we include `{}`s in these checks? I understand the value of the cleanliness of one-liners but I'm worried about the kinds of bugs that creep in due to oversight about missing brackets

Reply:
done